modules/core/src/convert_opt_fp16.cpp

```diff
@@ -0,0 +1,178 @@
+/*M///////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Please rename this file to "convert.fp16.cpp"

You can check effect via CMake output (`FP16 (0 files)`):
```
--     Dispatched code generation:  SSE4_1 FP16 AVX AVX2
--       requested:                 SSE4_1 AVX FP16 AVX2
--       SSE4_1 (0 files):          + SSE4_1
--       FP16 (0 files):            + SSE4_1 POPCNT SSE4_2 FP16 AVX
--       AVX (1 files):             + SSE4_1 POPCNT SSE4_2 AVX
--       AVX2 (1 files):            + SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2
```